File: productservice/src/main/java/com/productservice/models/Product.java
package com.productservice.models;

import java.math.BigDecimal;
import java.util.UUID;

import org.springframework.data.annotation.Id;

public class Product {
    @Id
    private int id;
    private String name;
    private String description;
    private BigDecimal price;

    public Product(String name, String description, BigDecimal price) {
        this.name = name;
        this.description = description;
        this.price = price;
    }

}

File: backend/search/SearchService/Models/Product.cs
﻿using System.Text.Json.Serialization;

namespace SearchService.Models;

public class Product
{
    [JsonPropertyName("id")]
    public Guid Id { get; set; } 
    [JsonPropertyName("name")]

    public string Name { get; set; } = null!;
    [JsonPropertyName("description")]

    public string Description { get; set; } = null!;
    [JsonPropertyName("price")]

    public decimal Price { get; set; }
}